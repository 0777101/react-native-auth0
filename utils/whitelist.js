export function apply(rules, values) {
  let mapped = {};
  for (let key of Object.keys(rules)) {
    let rule = rules[key];
    let value = values[key];
    if (value) {
      mapped[rule.toName || key] = value;
    } else if (rule.required) {
      throw new Error(rule.message || `The parameter ${key} is required`);
    }
  }
  return mapped;
}