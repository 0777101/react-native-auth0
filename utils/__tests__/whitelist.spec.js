import { apply } from '../whitelist'
import faker from 'faker';

describe('whitelist', () => {

  const rules = {
    state: {required: true, message: 'must have a state'},
    nonce: {required: false, message: 'must have a nonce'},
    clientId: {required: false, message: 'must have a clientId', toName: 'client_id'},
  };

  it('should keep expected value', () => {
    const value = { state: faker.random.uuid() };
    expect(apply(rules, value)).toMatchObject(value);
  });

  it('should fail if required key is not found', () => {
    const value = { state: faker.random.uuid() };
    expect(() => apply(rules, {})).toThrowErrorMatchingSnapshot();
  });

  it('should fail with default message', () => {
    const value = { state: faker.random.uuid() };
    expect(() => apply({state: {required: true}}, {})).toThrowErrorMatchingSnapshot();
  });

  it('should handle multiple parameters', () => {
    const value = {
      state: faker.random.uuid(),
      nonce: faker.random.uuid()
    };
    expect(apply(rules, value)).toMatchObject(value);
  });

  it('should ignore non declared keys', () => {
    const state = faker.random.uuid()
    const value = {
      state,
      connection: faker.random.uuid()
    };
    expect(apply(rules, value)).toMatchObject({state});
  });

  it('should use mapped name if available', () => {
    const state = faker.random.uuid()
    const clientId = faker.random.uuid()
    const value = {
      state,
      clientId
    };
    expect(apply(rules, value)).toMatchObject({
      state,
      'client_id': clientId
    });
  });
});