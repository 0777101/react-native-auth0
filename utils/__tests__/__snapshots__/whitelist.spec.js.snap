// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks should fail if required key is not found 1`] = `"must have a state"`;

exports[`checks should fail with default message 1`] = `"The parameter state is required"`;
